File: Year-2022/LeetCode-October-Challenge/AllPathsFromSourcetoTarget.js
var allPathsSourceTarget = function(graph) {
    let size = graph.length - 1;
    let temp = [], res = [];
    const DFS = (graph, currNode = 0) => {
        temp.push(currNode);
        if(currNode === size) res.push(Array.from(temp));
        for(let child of graph[currNode]){
            DFS(graph, child);
        }
        temp.pop();
    }
    DFS(graph);
    return res;
};

File: Year-2022/LeetCode-October-Challenge/AllPathsFromSourcetoTarget.java
import java.util.*;
class Solution {
    public int size;
    List<Integer> temp = new ArrayList<>();
    List<List<Integer>> res = new ArrayList<>();
    public void DFS(int[][] graph, int currNode){
        temp.add(currNode);
        if(currNode == size) res.add(new ArrayList<>(temp));
        for(int child: graph[currNode]){
            DFS(graph, child);
        }
        temp.remove(temp.size() - 1);
    }
    public List<List<Integer>> allPathsSourceTarget(int[][] graph) {
        size = graph.length - 1;
        DFS(graph, 0);
        return res;
    }
}
public class AllPathsFromSourcetoTarget {
    public static void main(String[] args) {
        
    }
}
